src/github.com/cloudfoundry-incubator/switchboard/runner/statuslogger/status_logger.go

```diff
@@ -0,0 +1,159 @@
+package statuslogger
+
+import (
+	"os"
+	"sync"
+	"time"
+
+	"code.cloudfoundry.org/lager/v3"
+	"github.com/tedsuo/ifrit"
+
+	"github.com/cloudfoundry-incubator/switchboard/domain"
+	"github.com/cloudfoundry-incubator/switchboard/runner/monitor"
+)
+
+type StatusLogger struct {
+	logger      lager.Logger
+	backends    []*domain.Backend
+	monitor     *monitor.ClusterMonitor
+	interval    time.Duration
+	backendChan chan *domain.Backend
+
+	// Protected by mutex
+	activeBackend    *domain.Backend
+	lastFailoverTime time.Time
+	lastFailoverFrom *domain.Backend
+	failoverCount    int
+	backendMutex     sync.RWMutex
+}
+
+func NewStatusLogger(
+	backends []*domain.Backend,
+	monitor *monitor.ClusterMonitor,
+	interval time.Duration,
+	logger lager.Logger,
+) *StatusLogger {
+	return &StatusLogger{
+		logger:      logger,
+		backends:    backends,
+		monitor:     monitor,
+		interval:    interval,
+		backendChan: make(chan *domain.Backend, 1),
+	}
+}
+
+// ActiveBackendChan returns the channel for receiving active backend updates.
+// This is primarily exposed for testing purposes.
+func (s *StatusLogger) ActiveBackendChan() chan *domain.Backend {
+	return s.backendChan
+}
+
+func (s *StatusLogger) Run(signals <-chan os.Signal, ready chan<- struct{}) error {
+	s.logger.Info("Status logger starting", lager.Data{
+		"interval":      s.interval.String(),
+		"backend_count": len(s.backends),
+	})
+
+	done := make(chan struct{})
+	shutdown := make(chan struct{})
+
+	// Register to receive active backend notifications
+	s.monitor.RegisterBackendSubscriber(s.backendChan)
+
+	// Listen for active backend changes and track failovers
+	go func() {
+		defer close(done)
+		for {
+			select {
+			case newActiveBackend := <-s.backendChan:
+				s.backendMutex.Lock()
+
+				// Track failover if backend changed
+				if s.activeBackend != newActiveBackend && s.activeBackend != nil {
+					now := time.Now()
+					s.lastFailoverTime = now
+					s.lastFailoverFrom = s.activeBackend
+					s.failoverCount++
+				}
+
+				s.activeBackend = newActiveBackend
+				s.backendMutex.Unlock()
+			case <-shutdown:
+				return
+			}
+		}
+	}()
+
+	ticker := time.NewTicker(s.interval)
+	defer ticker.Stop()
+
+	close(ready)
+
+	for {
+		select {
+		case <-signals:
+			s.logger.Info("Status logger received shutdown signal")
+			close(shutdown) // Signal goroutine to exit
```

Comment:
I like the comments here - these lines are always really hard to test to it's good to self-document how this will trigger the message on the done channel in the go routine up above to get past the blocking line below